internal/build/resolver/options/options.go

```diff
@@ -0,0 +1,88 @@
+package options
+
+import (
+	"context"
+
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/go-git/go-git/v5"
+)
+
+// OptionsResolverFn is a function for adding a branch filter to REST API requests
+type OptionsResolverFn func(context.Context) OptionsFn
+
+// OptionsFn is a function to apply modifications to the list builds API request ie. for adding additional filters
+type OptionsFn func(*buildkite.BuildsListOptions) error
+
+type AggregateResolver []OptionsResolverFn
+
+func (ar AggregateResolver) WithResolverWhen(condition bool, resovler OptionsResolverFn) AggregateResolver {
+	if condition {
+		return append(ar, resovler)
+	}
+	return ar
+}
+
+// ResolveBranchFromFlag returns a function that is used to add a branch filter to a build list options
+func ResolveBranchFromFlag(branch string) OptionsResolverFn {
+	return func(context.Context) OptionsFn {
+		return func(options *buildkite.BuildsListOptions) error {
```

Comment:
I'll get used to this but this feels super hard to glance eval.

Reply:
A function that returns a function that returns a function.

Reply:
I've probably done something very wrong

Reply:
Just pushed that change, what do you think now?